src/Build/BuildCheck/Infrastructure/BuildCheckManagerProvider.cs

```diff
@@ -503,8 +501,8 @@ public void FinalizeProcessing(LoggingContext loggingContext)
 
         private readonly ConcurrentDictionary<int, string> _projectsByInstanceId = new();
         private readonly ConcurrentDictionary<int, string> _projectsByEvaluationId = new();
-
-        private readonly ConcurrentDictionary<int, HashSet<string>> _deferredProjectEvalIdToImportedProjects = new();
+        // We are receiving project imported data only from the logger events - hence always in a single threaded context
```

Comment:
I couldn't find my answer in the infrastructure so I'm sorry but I will go ahead and ask.
Where do we guarantee that our events are synchronous please? Can't the logger get hit by two separate events, that will require handling at the same time? Or are these enqueued so the synchronization is guaranteed?
(The core of my question is that I see event, I think async. So this sentence goes against my intuition.)

Reply:
Added to the comment.